File: src/main/java/dev/schoenberg/evergore/protocolParser/rest/controller/api/wire/AvatarSummary.java
package dev.schoenberg.evergore.protocolParser.rest.controller.api.wire;

import java.time.Instant;

import com.fasterxml.jackson.annotation.JsonProperty;

public record AvatarSummary(@JsonProperty("avatar") String avatar, @JsonProperty("bankWithdrawn") long bankWithdrawn, @JsonProperty("bankDeposited") long bankDeposited,
		@JsonProperty("storageWithdrawn") long storageWithdrawn, @JsonProperty("storageDeposited") long storageDeposited, @JsonProperty("net") long net,
		@JsonProperty("donation") Long donation, @JsonProperty("craftSubsidy") Long craftSubsidy, @JsonProperty("lastBankActivity") Instant lastBankActivity,
		@JsonProperty("lastStorageActivity") Instant lastStorageActivity, @JsonProperty("staleSumsFrom") Instant staleSumsFrom) {}

File: frontend/src/domain/avatarSummary.ts
import type { Page } from './page.ts'

export interface AvatarSummary {
  readonly avatar: string
  readonly bankWithdrawn: number
  readonly bankDeposited: number
  readonly storageWithdrawn: number
  readonly storageDeposited: number
  readonly net: number
  readonly donation: number | null
  readonly craftSubsidy: number | null
  readonly lastBankActivity: Date | null
  readonly lastStorageActivity: Date | null
  readonly staleSumsFrom: Date | null
}

export interface GuildTotals {
  readonly bankWithdrawn: number
  readonly bankDeposited: number
  readonly storageWithdrawn: number
  readonly storageDeposited: number
  readonly net: number
  readonly donation: number | null
  readonly craftSubsidy: number | null
  readonly containsStaleSums: boolean
}

export interface Overview extends Page<AvatarSummary> {
  readonly totals: GuildTotals
}
